### ISSUE TITLE
cd <tab> should only print out paths

### ISSUE THREAD

**Issue by anntzer** [ipython/ipython#8894]
Currently, `cd <tab>` seems to autocomplete on everything (paths, globals, magics).  Note that this only happens if <tab> is pressed right after `cd`.

**Comment by Achifaifa** [comment]
I just tried this and it only autocompletes to paths

![](https://i.imgur.com/8FtMMUW.png)

**Comment by minrk** [comment]
I only see path completions, as well. Can you share your IPython config: `get_ipython().config`?

**Comment by jorisvandenbossche** [comment]
I also see this (IPython 4.0, Windows 7). But to specify, it only does this in certain directories (eg my main Users directory), and in other directories it only does it if there are no directories to complete.

**Comment by minrk** [comment]
@jorisvandenbossche thanks, that appears to be the key - other completions are allowed when the directory completions list is empty.

**Comment by anntzer** [comment]
Indeed, I also see this in my $HOME but not in other directories (afaict).
get_ipython().config contains

**Comment by minrk** [comment]
I see it in home, as well. Two cases, then.

**Comment by liushapku** [comment]
I also get this problem after upgrading to ipython6.1.0. Previously I used ipython5.3 and didn't see this problem. A screenshot is as below
![ipython6_complete_dir](https://user-images.githubusercontent.com/12460646/27669064-6b4bc93c-5cb7-11e7-87bd-43d7f8f26927.png)
The same occurs even if I was assigning an dir name to a variable, like this (note that some of the dir names are missing from the previous list)
![ipython6_2](https://user-images.githubusercontent.com/12460646/27669118-c65df52a-5cb7-11e7-9a27-76c94d288d23.png)
However, if I don't assign it to a variable, it behaves correctly (if I omit the starting quote, it is like the first screenshot):
![ipython6_3](https://user-images.githubusercontent.com/12460646/27669164-044d1d16-5cb8-11e7-9908-87d421e87f40.png)
